src/otaclient/create_standby/resume_ota.py

```diff
@@ -84,29 +85,31 @@ def _process_resource_at_thread(
 
             calculated_digest = hash_f.digest()
             if calculated_digest != expected_digest:
-                with contextlib.suppress(Exception):
-                    os.unlink(expected_digest_hex, dir_fd=self._resource_dir_fd)
-            else:
-                try:
-                    self._all_resource_digests.pop(calculated_digest)
-                    self._status_report_queue.put_nowait(
-                        StatusReport(
-                            payload=UpdateProgressReport(
-                                operation=UpdateProgressReport.Type.PREPARE_LOCAL_COPY,
-                                processed_file_num=1,
-                                processed_file_size=file_size,
-                            ),
-                            session_id=self.session_id,
-                        )
+                self._remove_entry(expected_digest_hex, dir_fd)
+                return
+
+            try:
+                self._all_resource_digests.pop(calculated_digest)
+                self._status_report_queue.put_nowait(
+                    StatusReport(
+                        payload=UpdateProgressReport(
+                            operation=UpdateProgressReport.Type.PREPARE_LOCAL_COPY,
+                            processed_file_num=1,
+                            processed_file_size=file_size,
+                        ),
+                        session_id=self.session_id,
                     )
-                except KeyError:
-                    with contextlib.suppress(Exception):
-                        os.unlink(expected_digest_hex, dir_fd=self._resource_dir_fd)
+                )
+            # basically should not happen, as now we only scan resources presented in
+            #   the target OTA image now.
+            except KeyError:
+                pass
         finally:
             self._se.release()
 
     def resume_ota(self) -> None:
         """Scan the OTA resource folder leftover by previous interrupted OTA."""
+        _resource_dir_fd = os.open(self._resource_dir, os.O_RDONLY)
```

Comment:
Not bind the opened resource_dir fd to class according to sonarcloud's suggestion.